### PR TITLE
[New] Add firmin@1.0.0 w/ git auto-update

### LINKED ISSUE #8181: [Request] Add firmin
**Library name:** firmin
**Git repository url:**https://github.com/beastaugh/firmin
**npm package url(optional):** 
**License(s):**BSD
**Official homepage:**http://extralogical.net/projects/firmin/
**Wanna say something? Leave message here:**
# 

Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sashberd**
close #8181, cc @beastaugh, using web-minify-helper tool for minifing

Pull request for issue: #8181
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/beastaugh/firmin
 * Official website (optional, not the repository): http://extralogical.net/projects/firmin
 * NPM package url (optional):
 * License and its reference: BSD-3-Clause
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 6
   - Count of stars: 200
   - Count of forks: 14
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@PeterDaveHello The files were updated. FYR

File: ajax/libs/firmin/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "firmin",
+  "filename": "firmin-1.0.0-min.js",
+  "description": "JavaScript animation library based on CSS transforms and transitions.",
```
Comment (sufuf3) [inline ajax/libs/firmin/package.json]: Let's add a `A` in the description, the whole line is `A JavaScript animation library using CSS transforms and transitions.`
(As the same with upstream's description in top of GitHub repo.)
Many thanks

File: ajax/libs/firmin/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "firmin",
+  "filename": "firmin-1.0.0-min.js",
```
Comment (sufuf3) [inline ajax/libs/firmin/package.json]: Would you please revise to `firmin.min.js`?
Many thanks.

**Comment by sashberd** [comment]
@sufuf3 The PR was updated.
See comment of @PeterDaveHello above about official release of version 1.0.0
Please close between you which files should be used

**Comment by sufuf3** [comment]
@sashberd Thanks.

@PeterDaveHello May I ask some questions?
1. As your mention https://github.com/cdnjs/cdnjs/pull/12095#pullrequestreview-89317104, this PR should add v1.0.0 from https://github.com/beastaugh/firmin/tree/master/site/downloads. Then, should we update .gitignore? because the auto-update still can work for adding the old versions.
2. Should we update filename from `firmin-1.0.0-min.js` to `firmin.min.js`, also rename the files' filename?
Many thanks

**Comment by PeterDaveHello** [comment]
Sorry for the late reply.

1. The concept to ignore or not, it depends on if the versions will be fetched or not, and if the versions are useful or not.

For example, if version `x.y.z` will be grabbed, but it's not useful, then we'll ignore it.
If the version `a.b.c` will be fetched, but as you'll need to manually change the file naming, which means you'll manually add those versions, then we don't need to ignore it in the `.gitignore`

2. Yes, we need to remove the versioning in the naming :+1:

**Comment by sashberd** [comment]
@PeterDaveHello @sufuf3 I did not get what changes I should do?

**Comment by sashberd** [comment]
@sufuf3 @extend1994 The PR was updated. FYR

**Comment by PeterDaveHello** [comment]
@sashberd must be something wrong in any steps ...

**Comment by PeterDaveHello** [comment]
@sashberd I thought you fixed the crlf issue in other PRs, let's fix it on this one? Though I can do it for you, but the key point is that we need to know the cause, and prevent it appears again in the future, what do you think? Thanks for your contribution again.

**Comment by sashberd** [comment]
@PeterDaveHello I actually did fix from my Mac, and I check it now with re-add files to the folder and there are no divergences between existed files in this PR and re-added files. I will send PR again to update, but I am not sure if this one will help.

**Comment by PeterDaveHello** [comment]
@sashberd did you grab the file freshly? Because this one, is actually not the crlf issue, but you manually added a final newline char there ... we shouldn't touch the content, remember? And git or the OS won't cause this problem, I bet you edit the file once ...

**Comment by sashberd** [comment]
@PeterDaveHello I swear you I even did not open this file, just download it with curl and rename it to without version number.
I am not a linux master, thus I have no idea  what commands are written inside link you attached. I will try to re-add files again, but...

**Comment by PeterDaveHello** [comment]
@sashberd can you give me the url you use with curl exactly so that I can help verify the issue?

**Comment by sashberd** [comment]
@PeterDaveHello curl https://raw.githubusercontent.com/beastaugh/firmin/master/site/downloads/firmin-1.0.0-min.js -O

**Comment by PeterDaveHello** [comment]
@sashberd I mean "firmin.js", the file with the problem.

**Comment by PeterDaveHello** [comment]
@sashberd https://raw.githubusercontent.com/beastaugh/firmin/master/site/downloads/firmin-1.0.0.js  ?

**Comment by PeterDaveHello** [comment]
@sashberd if so, it's obviously you grabbed the wrong, it's not about linux skills or something realted. That's why I suggest you to run auto-updater locally.

The url you should use is the one in 1.0.0 tag https://raw.githubusercontent.com/beastaugh/firmin/1.0.0/src/firmin.js, not the one in master branch https://raw.githubusercontent.com/beastaugh/firmin/master/site/downloads/firmin-1.0.0.js, we've discussed before.

Correct me please if I'm wrong.

**Comment by sashberd** [comment]
@PeterDaveHello You told us to use those files [here](https://github.com/cdnjs/cdnjs/pull/12095#pullrequestreview-89317104)

**Comment by PeterDaveHello** [comment]
Ouch, I forgot that, my fault.
As this is the special case, that info should be added in the commit message, so everyone can follow up, and won't miss it, would you please update the commit message and I'll merge it? Many thanks.

**Comment by PeterDaveHello** [comment]
(Just mention the files of v1.0.0 are grabbed from another URL)

**Comment by sashberd** [comment]
@PeterDaveHello The commit message was updated. FYR

**Pull request merged**